src/platform/src/main/java/org/geoserver/platform/util/Disposer.java

```diff
@@ -0,0 +1,238 @@
+     */
+    public <T extends DisposableBean> T replace(Object oldBean, T newBean) {
+        return replace(oldBean, newBean, DESTROYER_OF_BEANS);
+    }
+    
+    /**
+     * Register an AutoCloseable for disposal.
+     * @param obj AutoClosable to register
+     */
+    public void register(AutoCloseable obj) {
+        register(obj, DESTROYER_OF_CLOSEABLES);
+    }
+    
+    /**
+     * Dispose of an object if it is registered then register an AutoCloseable in its place.
+     * @param oldObj Object to replace.  May be null.
+     * @param newObj Object to register
+     * @return the new object
+     */
+    public <T extends AutoCloseable> T replace(Object oldObj, T newObj) {
+        return replace(oldObj, newObj, DESTROYER_OF_CLOSEABLES);
+    }
+    
+    /**
+     * Register an ExecutorService for disposal.
+     * @param obj ExecutorService to register
+     */
+    public void register(ExecutorService obj) {
+        register(obj, DESTROYER_OF_EXECUTORS);
+    }
+    
+    /**
+     * Dispose of an object if it is registered then register an ExecutorService in its place.
+     * @param oldObj Object to replace.  May be null.
+     * @param newObj Object to register
+     * @return the new object
+     */
+    public <T extends ExecutorService> T replace(Object oldObj, T newObj) {
+        return replace(oldObj, newObj, DESTROYER_OF_EXECUTORS);
+    }
+
+    @SuppressWarnings("unchecked") // The types should have matched up when they were registered
+    private boolean doDestroy(Object obj, Destroyer<?> dest) {
+        try {
+            ((Destroyer<Object>)dest).accept(obj);
+            return true;
+        } catch (Exception e) {
+            LOGGER.log(Level.SEVERE, "Error while disposing of registered disposable object", e);
+            return false;
+        }
+    }
+    
+    /**
+     * Dispose of all registered objects.  Once this method has been called, all other public 
+     * methods will immediately dispose of any arguments given to them and then throw 
+     * IllegalStateException.
+     * <p>
+     * This method is idempotent. Calling it again will have no additional effect.
+     */
+    @Override
+    public void destroy() {
+        if(toDestroy==null) return;
+        synchronized(this) {
+            if(toDestroy==null) return;
+            for(Map.Entry<Object, Destroyer<?>> pair : toDestroy.entrySet()) {
+                doDestroy(pair.getKey(), pair.getValue());
+            }
+            toDestroy = null;
+        }
+    }
+    
+    private static class DestroyerDisposedException extends IllegalArgumentException {
+        /** serialVersionUID */
+        private static final long serialVersionUID = 4547015737240523972L;
+
+        DestroyerDisposedException(){
+            super("DestroyerBean has been destroyed.");
+        }
+    }
+    
+    private void checkNotDestroyed() throws IllegalArgumentException {
+        if(toDestroy==null) throw new DestroyerDisposedException();
```

Comment:
Does this respect our coding conventions? If my memory serves me right we should use {} right?